src/Domain/IndexManagement/IndexConfiguration.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JeroenG\Explorer\Domain\IndexManagement;
+
+class IndexConfiguration
+{
+    private string $name;
+
+    private array $settings = [];
+
+    private array $properties = [];
+
+    public static function empty(string $name): self
+    {
+        $config = new self();
+        $config->name = $name;
+        return $config;
+    }
+
+    public static function create(string $name, array $properties, array $settings): self
```

Comment:
Should this mean the constructor should be private (to disallow using that to create an incorrect object)?

Reply:
It should and it would